File: Java/com/gildedrose/GildedRoseTest.java
package com.gildedrose;

public class GildedRoseTest {
    public static void main(String[] args) {
        testConjuredBeforeSellDate();
        testConjuredAfterSellDate();
    }

    private static void testConjuredBeforeSellDate() {
        Item[] items = new Item[] { new Item("Conjured Mana Cake", 5, 10) };
        GildedRose app = new GildedRose(items);
        app.updateQuality();

        if (items[0].quality == 8) {
            System.out.println("testConjuredBeforeSellDate passed");
        } else {
            System.out.println(" testConjuredBeforeSellDate failed: expected 8, got " + items[0].quality);
        }
    }

    private static void testConjuredAfterSellDate() {
        Item[] items = new Item[] { new Item("Conjured Mana Cake", 0, 10) };
        GildedRose app = new GildedRose(items);
        app.updateQuality();

        if (items[0].quality == 6) {
            System.out.println(" testConjuredAfterSellDate passed");
        } else {
            System.out.println(" testConjuredAfterSellDate failed: expected 6, got " + items[0].quality);
        }
    }
}

File: scala/src/test/scala/com/gildedrose/GildedRoseTest.scala
package com.gildedrose

import org.scalatest._

class GildedRoseTest  extends FlatSpec with Matchers {
      it should "foo" in {
        var items = Array[Item](new Item("foo", 0, 0))
        val app = new GildedRose(items)
        app.updateQuality()
        (app.items(0).name) should equal ("fixme")
      }
}

  test("Conjured items degrade by 2 before sell date") {
    val items = Array(new Item("Conjured Mana Cake", 5, 10))
    val app = new GildedRose(items)
    app.updateQuality()
    assert(items(0).quality == 8)
  }

  test("Conjured items degrade by 4 after sell date") {
    val items = Array(new Item("Conjured Mana Cake", 0, 10))
    val app = new GildedRose(items)
    app.updateQuality()
    assert(items(0).quality == 6)
  }

